src/transformation/mesh_intersection/mesh_intersection.rs

```diff
@@ -143,33 +163,40 @@ pub fn intersect_meshes_with_tolerances(
     // 4: Initialize a new mesh by inserting points into a set. Duplicate points should
     // hash to the same index.
     let mut point_set = RTree::<TreePoint, _>::new();
-    let mut topology_indices = Vec::new();
+    let mut topology_indices = HashMap::new();
```

Comment:
I am a bit scared of this. The non-determinism might create hard to reproduce bugs in the future, since the traversal order of the faces will change between invocations. I would like to suggest a BTreeMap. It won't necessarily be slower in most cases because the hashmap iteration will suffer from more cache misses.

Reply:
Agreed, I’ll switch to `crate::utils::hashmap::HashMap` which is deterministic.